packages/core/state/interaction/logics.ts

```diff
@@ -582,17 +582,81 @@ const setIsSmallScreen = createLogic({
  * Logs details of files that are being moved.
  */
 const moveFilesLogic = createLogic({
-    async process({ action, getState }: ReduxLogicDeps, _dispatch, done) {
+    async process({ action, getState }: ReduxLogicDeps, dispatch, done) {
         try {
             const httpFileService = interactionSelectors.getHttpFileService(getState());
             const username = interactionSelectors.getUserName(getState());
-            const fileIds = (action as MoveFilesAction).payload.fileDetails.map((file) => file.id);
-            const cacheStatuses = await httpFileService.cacheFiles(fileIds, username);
 
-            // TODO: What to do with the status
-            console.log("Cache statuses:", cacheStatuses);
+            const fileDetails = (action as MoveFilesAction).payload.fileDetails;
+
+            // Map file IDs to file names for easy lookup
+            const fileIdToNameMap = Object.fromEntries(
+                fileDetails.map((file) => [file.id, file.name])
+            );
+
+            // Extract file IDs
+            const fileIds = fileDetails.map((file) => file.id);
+
+            const response = await httpFileService.cacheFiles(fileIds, username);
+            const cacheStatuses = response.cacheFileStatuses;
+
+            // Check if the response is empty.
+            if (!cacheStatuses || Object.keys(cacheStatuses).length === 0) {
+                dispatch(
+                    interaction.actions.processError(
+                        "moveFilesNoFilesProcessed",
+                        "No files were processed. Please check the request or try again."
+                    )
+                );
+                return;
+            }
+
+            const successfulFiles: string[] = [];
+            const failedFiles: string[] = [];
+
+            Object.entries(cacheStatuses).forEach(([fileId, status]) => {
+                if (
+                    status === "DOWNLOAD_COMPLETE" ||
+                    status === "DOWNLOAD_IN_PROGRESS" ||
+                    status === "DOWNLOAD_STARTED"
+                ) {
+                    successfulFiles.push(fileId);
+                } else if (
+                    status === "FILE_RECORD_NOT_FOUND" ||
+                    status === "FILE_NOT_FOUND_ON_CLOUD"
+                ) {
+                    failedFiles.push(fileId);
+                }
```

Comment:
Are there other statuses possible? I wonder if it would make more sense to just keep a list of "success" statuses and if the status isn't in that list then it is considered a failure

Reply:
These are all of the current status possibilities, as detailed[ here](https://github.com/AllenInstitute/biofile-finder/pull/322#issuecomment-2486626852). In the event of an unknown status, I don't think we should mark it as failed, as it could also be an alternate success. Perhaps a third option or warning?